server/user.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * @fileoverview user function handler
+ */
+var fs = require('fs');
+var assert = require('assert');
+
+var log = require('./log.js');
+var utils = require('./utils.js');
+var database = require('./database.js');
+
+/** @type {user} */
+module.exports = user;
+
+/**
+ * @constructor
+ */
+function user() {}
+
+/** @enum {string} */
+user.QueryType = {
+  SIGNUP: 'sign-up',
+  SIGNIN: 'sign-in'
+};
+
+/** @const {RegExp} */
+user.VALID_EMAIL_REGEX =
+  /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+
+/** @const {RegExp} */
+user.VALID_USERNAME_REGEX = /^\w{6,}$/;
+
+/** @const {RegExp} */
+user.VALID_PASSWORD_REGEX = /^\w{8,}$/;
+
+/**
+ * Expire time of cookie.
+ * @const {number}
+ */
+user.cookieExpireTime = 24 * 60 * 60 * 1000;
+
+/**
+ * @typedef {{
+ *   email: (string|undefined),
+ *   username: string,
+ *   password: string,
+ *   confirmed: (boolean|undefined)
+ * }}
+ */
+user.Info;
+
+/**
+ * @typedef {{
+ *   error: string
+ * }}
+ */
+user.Error;
+
+/**
+ * Checks the user information for signing in.
+ * @param {!mongodb.Db} db Session database.
+ * @param {!user.Info} userInfo User Information.
+ * @param {function(!Object)} callback Callback function.
+ */
+user.signUp = function(db, userInfo, callback) {
+  var cursor = db.collection('userInfo').find({
```

Comment:
I think you mean {username = a or email = b}. 
You may also consider aborting on first error, rather than getting a list of all errors. The latter is less common in error handling routine. In this case it may not differ much though.